tests/storage/snapshot.go

```diff
@@ -834,6 +834,70 @@ var _ = Describe(SIG("VirtualMachineSnapshot Tests", func() {
 					}
 				})
 			})
+			It("should not include pending volumes", func() {
+				blankDv := func() *cdiv1.DataVolume {
+					return libdv.NewDataVolume(
+						libdv.WithNamespace(testsuite.GetTestNamespace(nil)),
+						libdv.WithBlankImageSource(),
+						libdv.WithStorage(
+							libdv.StorageWithStorageClass(snapshotStorageClass),
+							libdv.StorageWithVolumeSize(cd.BlankVolumeSize),
+						),
+						libdv.WithForceBindAnnotation(),
+					)
+				}
+
+				Expect(err).ToNot(HaveOccurred())
+				vmi := libvmifact.NewFedora(libnet.WithMasqueradeNetworking())
+				vmi.Namespace = testsuite.GetTestNamespace(nil)
+				vm = libvmi.NewVirtualMachine(vmi)
+
+				dv1 := blankDv()
+
+				libstorage.AddDataVolumeTemplate(vm, dv1)
+				libstorage.AddDataVolume(vm, dv1.Name, dv1)
+
+				vm, vmi = createAndStartVM(vm)
+				libwait.WaitForSuccessfulVMIStart(vmi,
+					libwait.WithTimeout(300),
+				)
+				Eventually(matcher.ThisVMI(vmi), 12*time.Minute, 2*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstanceAgentConnected))
+
+				vm, err := virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Adding an additional volume to the VM while it's online")
```

Comment:
maybe add to the msg or in a comment that it will be added only to the VM and not to the VMI since it is not a hotplug.